bento_wes/routers/runs/run_details.py

```diff
@@ -0,0 +1,104 @@
+from fastapi import APIRouter, Depends, Form
+from fastapi.responses import PlainTextResponse, FileResponse
+from fastapi.exceptions import HTTPException
+import shutil
+import urllib.parse
+from uuid import UUID
+from pathlib import Path
+
+from bento_lib.auth.permissions import P_VIEW_RUNS
+
+from bento_wes import states
+from bento_wes.db import DatabaseDep
+from bento_wes.types import RunStream
+from bento_wes.celery import celery
+from bento_wes.config import SettingsDep
+
+from .deps import stash_run_or_404, get_stream, RunDep, AuthzDep
+from .utils import _denest_list
+from .constants import RUN_CANCEL_BAD_REQUEST_STATES
+
+detail_router = APIRouter(prefix="/{run_id}")
+detail_router.dependencies.append(Depends(stash_run_or_404))
+
+
+@detail_router.get("")
+async def get_run(run: RunDep, authz_check: AuthzDep):
+    await authz_check(P_VIEW_RUNS, run.request.get_authz_resource())
+    return run.model_dump(mode="json")
+
+
+@detail_router.get("/status")
+async def run_status(run_id: UUID, run: RunDep, db: DatabaseDep, authz_check: AuthzDep):
+    await authz_check(P_VIEW_RUNS, run.request.get_authz_resource())
+
+    run = db.get_run(db.c, run_id)
+    return run.model_dump(mode="json")
+
+
+@detail_router.post("/download-artifact")
+async def run_download_artifact(
+    run_id: UUID,
+    run: RunDep,
+    authz_check: AuthzDep,
+    path: str = Form(...),
+):
+    await authz_check(P_VIEW_RUNS, run.request.get_authz_resource())
+
+    artifact_path = path.strip()
+    if not artifact_path:
+        raise HTTPException(status_code=400, detail="Requested artifact path is blank or unspecified")
```

Comment:
for all status codes in this file, use `fastapi.status` constants (see `https://fastapi.tiangolo.com/reference/status/`) for consistency with other parts of Bento and for descriptive names